Test suite fails as http://ip.jsontest.com is broken

## Reporting Bugs

http://ip.jsontest.com is broken, leading to test failures: (tested on commit b5c665edf41352cd1647d720eaf9e4199436f0e1)

```
$ pytest tests/test_core/test_responses_module.py::TestResponsesMockWithPassThru
=========================================================================== test session starts ============================================================================
platform linux -- Python 3.12.7, pytest-8.3.3, pluggy-1.5.0
rootdir: /home/yen/tmp/moto
configfile: pyproject.toml
collected 2 items

tests/test_core/test_responses_module.py FF                                                                                                                          [100%]

================================================================================= FAILURES =================================================================================
_________________________________________________________ TestResponsesMockWithPassThru.test_aws_and_http_requests _________________________________________________________

self = <tests.test_core.test_responses_module.TestResponsesMockWithPassThru testMethod=test_aws_and_http_requests>

    def test_aws_and_http_requests(self) -> None:
        self.r_mock.add(responses.POST, "https://example.org", status=200)
>       self.assertEqual("OK", self.aws_and_http_requests())

tests/test_core/test_responses_module.py:111:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
moto/core/models.py:122: in wrapper
    result = func(*args, **kwargs)
tests/test_core/test_responses_module.py:102: in aws_and_http_requests
    self.http_requests()
moto/core/models.py:122: in wrapper
    result = func(*args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <tests.test_core.test_responses_module.TestResponsesMockWithPassThru testMethod=test_aws_and_http_requests>

    def http_requests(self) -> str:
        # Mock this website
        requests.post("https://example.org")

        # Passthrough this website
>       assert requests.get("http://ip.jsontest.com").status_code == 200
E       AssertionError: assert 404 == 200
E        +  where 404 = <Response [404]>.status_code
E        +    where <Response [404]> = <function get at 0x72972f62c5e0>('http://ip.jsontest.com')
E        +      where <function get at 0x72972f62c5e0> = requests.get

tests/test_core/test_responses_module.py:95: AssertionError
_____________________________________________________________ TestResponsesMockWithPassThru.test_http_requests _____________________________________________________________

self = <tests.test_core.test_responses_module.TestResponsesMockWithPassThru testMethod=test_http_requests>

    def test_http_requests(self) -> None:
        self.r_mock.add(responses.POST, "https://example.org", status=200)
>       self.assertEqual("OK", self.http_requests())

tests/test_core/test_responses_module.py:107:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
moto/core/models.py:122: in wrapper
    result = func(*args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <tests.test_core.test_responses_module.TestResponsesMockWithPassThru testMethod=test_http_requests>

    def http_requests(self) -> str:
        # Mock this website
        requests.post("https://example.org")

        # Passthrough this website
>       assert requests.get("http://ip.jsontest.com").status_code == 200
E       AssertionError: assert 404 == 200
E        +  where 404 = <Response [404]>.status_code
E        +    where <Response [404]> = <function get at 0x72972f62c5e0>('http://ip.jsontest.com')
E        +      where <function get at 0x72972f62c5e0> = requests.get

tests/test_core/test_responses_module.py:95: AssertionError
============================================================================= warnings summary =============================================================================
tests/test_core/test_responses_module.py::TestResponsesMockWithPassThru::test_aws_and_http_requests
  /usr/lib/python3.12/site-packages/botocore/auth.py:424: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
    datetime_now = datetime.datetime.utcnow()

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================================================================= short test summary info ==========================================================================
FAILED tests/test_core/test_responses_module.py::TestResponsesMockWithPassThru::test_aws_and_http_requests - AssertionError: assert 404 == 200
FAILED tests/test_core/test_responses_module.py::TestResponsesMockWithPassThru::test_http_requests - AssertionError: assert 404 == 200
======================================================================= 2 failed, 1 warning in 0.47s =======================================================================
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,16 @@
 Moto Changelog
 ==============
 
+5.0.24
+-----
+Docker Digest for 5.0.24: <autopopulateddigest>
+
+    Miscellaneous:
+        * EC2: Terminating instances will now release private ip addresses from the NIC's attached to the interface
+        * S3: Fixes a bug in complete_multipart_upload() where it was no longer possible to overwrite an earlier multipart upload (Broken in 5.0.23)
+        * S3: get_object_cors() now correctly returns the ExposeHeader-value
+
+
 5.0.23
 -----
 Docker Digest for 5.0.23: _sha256:d41e007bb1f7d41b530959ae9cbed1edf42737ee839faf8da7e925bf19f63105_

--- a/tests/test_core/test_responses_module.py
+++ b/tests/test_core/test_responses_module.py
@@ -2,6 +2,9 @@
 Ensure that the responses module plays nice with our mocks
 """
 
+from http.server import BaseHTTPRequestHandler, HTTPServer
+from threading import Event, Thread
+from typing import Optional, Tuple
 from unittest import SkipTest, TestCase
 
 import boto3
@@ -67,6 +70,45 @@ class TestResponsesModule(TestCase):
             assert r.json() == {"a": "4"}
 
 
+class WebRequestHandler(BaseHTTPRequestHandler):
+    def do_GET(self) -> None:
+        self.send_response(200)
+        self.send_header("Content-Type", "application/json")
+        self.end_headers()
+        self.wfile.write(b"real response")
+
+
+class SimpleServer:
+    def __init__(self) -> None:
+        self._port = 0
+
+        self._thread: Optional[Thread] = None
+        self._ip_address = "0.0.0.0"
+        self._server: Optional[HTTPServer] = None
+        self._server_ready_event = Event()
+
+    def _server_entry(self) -> None:
+        self._server = HTTPServer(("0.0.0.0", 0), WebRequestHandler)
+        self._server_ready_event.set()
+        self._server.serve_forever()
+
+    def start(self) -> None:
+        self._thread = Thread(target=self._server_entry, daemon=True)
+        self._thread.start()
+        self._server_ready_event.wait()
+
+    def get_host_and_port(self) -> Tuple[str, int]:
+        assert self._server
+        host, port = self._server.server_address
+        return (str(host), port)
+
+    def stop(self) -> None:
+        if self._server:
+            self._server.shutdown()
+        if self._thread:
+            self._thread.join()
+
+
 @mock_aws
 class TestResponsesMockWithPassThru(TestCase):
     """
@@ -77,24 +119,29 @@ class TestResponsesMockWithPassThru(TestCase):
         if RESPONSES_VERSION < LooseVersion("0.24.0"):
             raise SkipTest("Can only test this with responses >= 0.24.0")
 
-        raise SkipTest("Endpoint that we're trying to reach is down")
+        self.server = SimpleServer()
+        self.server.start()
+        host, port = self.server.get_host_and_port()
+        self.server_url = f"http://{host}:{port}"
 
         self.r_mock = responses.RequestsMock(assert_all_requests_are_fired=True)
         override_responses_real_send(self.r_mock)
         self.r_mock.start()
-        self.r_mock.add_passthru("http://ip.jsontest.com")
+        self.r_mock.add_passthru(self.server_url)
 
     def tearDown(self) -> None:
         self.r_mock.stop()  # type: ignore
         self.r_mock.reset()  # type: ignore
         override_responses_real_send(None)
 
+        self.server.stop()
+
     def http_requests(self) -> str:
         # Mock this website
         requests.post("https://example.org")
 
         # Passthrough this website
-        assert requests.get("http://ip.jsontest.com").status_code == 200
+        assert requests.get(self.server_url).content == b"real response"
 
         return "OK"
 

